proxy/src/console/provider/neon.rs

```diff
@@ -277,6 +281,14 @@ impl super::Api for Api {
             return Ok(cached);
         }
 
+        // check rate limit
+        if !self
```

Comment:
It feels like it's pretty much the same as auth rate limiter:

I see two cases:
* If it's not a valid connection, than it will be ratelimited (after reaching some rps) 
* If it's a valid connection, at some point it will hit the Postgres limit and it would be an invalid connection.

In the end of the day, it is the same as doing auth rate limiter. So, why wouldn't we move this rate lmiter to the earlier step, potentially saving our CPU?

WDYT?

Reply:
I thought this was the solution we discussed wrt cache invalidation + connection failure feedback loops. Perhaps I misinterpreted what you meant yesterday

Reply:
No, you understand it correctly. This is just an idea, that I've got now while reading your PR.

It looks like moving rate limiting before the authentication would have the same effect on the `wake_compute` rate, because, proxy call cplane no more than once on each request. So, it would be the same, but moving it before the auth, also saves a bit of CPU during the auth.

Reply:
Ok, sounds good.